web/docs/operators/sources/context.md

```diff
@@ -0,0 +1,58 @@
+# context
+
+Creates or removes a context. The contexts defined via this source can be used
+with the [`context` transformation](../transformations/context.md).
+
+## Synopsis
+
+```
+context [action] <name>
+
+actions:
+  apply
+  create
+  remove
+  update
+```
+
+## Description
+
+The `context` operator manages and applies contexts. Contexts are a flexible
+mechanism for data enrichment. A context has a type and is implemented as a
+`context plugin`. Each context plugin provides a specific feature set. For
+example the `constants` context can be used to extend events with fields that
+are defined by the key-value-pairs passed to the `context create <name> --type
+constants` operator call.
+
+### `create`
+
+Creates a new context and emits an event that represents it's initial state.
+The schema of that event is specific to the context-type.
+
+#### Synopsis
+
+```
+context create <name> [key=value]
+
+options:
+  --type context-type   The context type to instantiate.
```

Comment:
It isn't it was just implement this way because of the design doc.

Reply:
Why is it optional in the design if this doesn't make sense? Well, anyway, I guess that means that we can change this if we want to. A required named argument would also be a possibility, but normally those can be moved around, which does not seem to be the case here.

Reply:
I think the document was still a bit rough and brainstormy at the time I wrote the parser. Obviously you need a type when you create a context, we need to adjust the documentation and operator interface accordingly.

Reply:
This error comes from the empty optional deref, but hanks for fixing this :+1:

I still think that this should probably be a positional argument, because:

> A required named argument would also be a possibility, but normally those can be moved around, which does not seem to be the case here.